tests/test_main.py

```diff
@@ -130,3 +130,23 @@ def test_error_message_environment_file_not_found(capsys, tmpdir, explicit_file,
     out, err = capsys.readouterr()
     assert out == ''
     assert err == 'File "{0}" does not exist.\n'.format(str(tmpdir / expected_name))
+
+
+def test_get_env_directory(mocker):
+    import os
+    mock_output = b'''
+        {
+        "envs": ["W:\\\\0\\\\envs\\\\my_env", "W:\\\\1\\\\envs\\\\my_env"],
```

Comment:
I suggest to use `tmpdir` to actually create the directories. This way you only need to mock `subprocess.check_output` and doesn't have to juggle around the path separators so much. 😉

Reply:
Something like (untested):

```python
def test_get_env_directory(mocker, tmpdir):
    import os
    env_0 = tmpdir.join('0/envs/my_env').ensure(dir=1)
    env_1 = tmpdir.join('1/envs/my_env').ensure(dir=1)
    mock_output = b'''
        {
        "envs": [str(env_0), str(env_1)],
        "envs_dirs": [str(tmpdir.join('1/envs'))]
        }
    '''
    mocker.patch('subprocess.check_output', return_value=mock_output)
    
    obtained = devenv.get_env_directory('my_env')    
    assert obtained == str(env_1)
    
    env_1.remove()    
    assert devenv.get_env_directory('my_env') is None
```

Also please add a docstring explaining the purpose of the test.

Reply:
Remember that `conda-meta` sub folder is also necessary.

Reply:
Right, thanks